python/tests/unit/arcticdb/version_store/test_sort_merge.py

```diff
@@ -135,35 +136,29 @@ def test_merge_strings_dynamic(lmdb_library):
     expected_df = pd.DataFrame(expected_values, index=expected_dates)
     assert_frame_equal(lib.read(sym1).data, expected_df)
 
-
-@pytest.mark.xfail(reason="Unsorted segments are not implemented")
-def test_unordered_segment(lmdb_library):
-    lib = lmdb_library
+def test_unordered_segment(lmdb_library_static_dynamic):
+    lib = lmdb_library_static_dynamic
     dates = [np.datetime64('2023-01-03'), np.datetime64('2023-01-01'), np.datetime64('2023-01-05')]
     df = pd.DataFrame({"col": [2, 1, 3]}, index=dates)
     lib.write("sym", df, staged=True)
     lib.sort_and_finalize_staged_data("sym")
-    assert_frame_equal(lib.read('sym'), pd.DataFrame({"col": [1, 2, 3]}, index=[np.datetime64('2023-01-01'), np.datetime64('2023-01-03'), np.datetime64('2023-01-05')]))
+    assert_frame_equal(lib.read('sym').data, pd.DataFrame({"col": [1, 2, 3]}, index=[np.datetime64('2023-01-01'), np.datetime64('2023-01-03'), np.datetime64('2023-01-05')]))
 
-def test_repeating_index_values(lmdb_library):
-    lib = lmdb_library
+def test_repeating_index_values(lmdb_library_static_dynamic):
+    lib = lmdb_library_static_dynamic
     dates = [np.datetime64('2023-01-01'), np.datetime64('2023-01-03'), np.datetime64('2023-01-05')]
     df1 = pd.DataFrame({"col": [1,2,3]}, index=dates)
     df2 = pd.DataFrame({"col": [4,5,6]}, index=dates)
     lib.write("sym", df1, staged=True)
     lib.write("sym", df2, staged=True)
     lib.sort_and_finalize_staged_data("sym")
-    expected_index = pd.DatetimeIndex([np.datetime64('2023-01-01'), np.datetime64('2023-01-01'), np.datetime64('2023-01-03'),
-                                       np.datetime64('2023-01-03'), np.datetime64('2023-01-05'), np.datetime64('2023-01-05')],
-                                      dtype="datetime64[ns]")
     data = lib.read("sym").data
-    assert data.index.equals(expected_index)
-    for i in range(0, len(data["col"])):
-        assert data["col"][i] == df1["col"][i // 2] or data["col"][i] == df2["col"][i // 2]
+    expected = pd.concat([df1, df2]).sort_index()
+    assert data.index.equals(expected.index)
```

Comment:
We can also check that the values in `col` are correct. First value is 1 or 4, second value is 1 or 4 and not equal to the first value, etc